### PR TITLE
EAP7-1159: Support runtime operations for messaging servers

### PR DESCRIPTION
**Pull request by hpehl**
https://issues.jboss.org/browse/EAP7-1159

### CODE REVIEW

File: console/EAP7-1159_Support_runtime_operations_for_messaging_servers.adoc [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+=== Affected Projects or Components
+
+* HAL
+
+== Requirements
+
+The console adds the following actions to the messaging server runtime column:
+
+* View: Will open a new application screen (see below)
+* `export-journal()`
```
Comment (jmesnil) [inline console/EAP7-1159_Support_runtime_operations_for_messaging_servers.adoc]: this operation will create a XML file on the server file system (in the same directory where the journal log files are stored)

File: console/EAP7-1159_Support_runtime_operations_for_messaging_servers.adoc [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+* `import-journal(String xmlFile)`: Does this operation support file uploads?
```
Comment (jmesnil) [inline console/EAP7-1159_Support_runtime_operations_for_messaging_servers.adoc]: I am not sure, the file parameter is flagged with `"filesystem-path" => true`

File: console/EAP7-1159_Support_runtime_operations_for_messaging_servers.adoc [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+* `force-failover()`
+* `reset-all-message-counter-histories()`
+* `reset-all-message-counters()`
+
+The new application screen will show various information about the selected messaging server. The following operations will be supported:
+
+=== Connections
+
+* `list-connection-ids()`
+* `list-connections-as-json()`
```
Comment (jmesnil) [inline console/EAP7-1159_Support_runtime_operations_for_messaging_servers.adoc]: for historical reasons, some management operations have 2 variants, one with a DMR output, the other with JSON (and sometimes HTML).

I don't think it is necessary to expose both to HAL. Exposing the DRM-variant is enough

File: console/EAP7-1159_Support_runtime_operations_for_messaging_servers.adoc [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+* `list-prepared-transaction-details-as-html()`
+* `list-prepared-transaction-details-as-json()`
+* `list-prepared-transaction-jms-details-as-html()`
+* `list-prepared-transaction-jms-details-as-json()`
+* `list-prepared-transactions()`
+
+=== Unknown Category
+
+* `get-address-settings-as-json(String addressMatch)`
+* `list-remote-addresses(String ipAddress)`
```
Comment (jmesnil) [inline console/EAP7-1159_Support_runtime_operations_for_messaging_servers.adoc]: that one could go with the "Connections" tab

File: console/EAP7-1159_Support_runtime_operations_for_messaging_servers.adoc [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+* `resolve-address-setting(String activeMQAddress)`
```
Comment (jmesnil) [inline console/EAP7-1159_Support_runtime_operations_for_messaging_servers.adoc]: get-address-settings-as-json and resolve-address-setting can be grouped together in an "Adress Settings" tab. These operations are purely informational (and really for debugging purpose) to figure out which settings applies to a given Artemis address.

File: console/EAP7-1159_Support_runtime_operations_for_messaging_servers.adoc (hunk 3 above)
Comment (hpehl) [reply console/EAP7-1159_Support_runtime_operations_for_messaging_servers.adoc]: I cannot find an equivalent DMR-variant. `list-connection-ids()` lists only the connection IDs, whereas `list-connections-as-json()` returns more details like `clientAddress` or `sessionCount`.

**Comment by jamezp** [comment]
@jmesnil Does this seem okay now?

**Pull request merged**